### ISSUE TITLE
Update the balance_lib to use the latest version

### ISSUE THREAD

**Issue by joshua-augustinus** [beyond-all-reason/balance_algorithm#3]
The balance_lib.ex in this tester project needs to be updated. I have copied the version from here:
https://github.com/beyond-all-reason/teiserver

and placed it in this project. I also noticed that the function to get captain is wrong and I have fixed it. The captain of the team is the highest rated member on the team.

## How to test
Run the commands in the readme and all tests should pass

**Comment by joshua-augustinus** [comment]
I had a chat with @Teifion on how to get this stuff merged. He said to ping @Beherith and he would know. I get that you're probably busy with the Event but just wanted to put this on your radar and perhaps redirect me to other Elixir devs (if there are any). The purpose of this PR is to get the test framework up to date to unblock larger changes in the future.

BTW I bumped into Teifion after he responded to my [Elixir question](https://elixirforum.com/t/return-type-of-map/61892/5) and I chatted with him privately.
